kurtosis/kurtosis/servicesavalanche/avalanchegonode/avalanchego_node_api_service.go

```diff
@@ -12,35 +12,38 @@ import (
 )
 
 type NodeAPIService struct {
-	serviceID          services.ServiceID
-	ipAddr             string
+	serviceCtx		   *services.ServiceContext
 	httpPort           int
 	stakingPort        int
 	bootstrappedPChain bool
 	bootstrappedCChain bool
 	bootstrappedXChain bool
 }
 
-func NewNodeAPIService(serviceID services.ServiceID, ipAddr string, httpPort int, stakePort int) *NodeAPIService {
-	return &NodeAPIService{serviceID: serviceID, ipAddr: ipAddr, httpPort: httpPort, stakingPort: stakePort}
+func NewNodeAPIService(serviceCtx *services.ServiceContext, httpPort int, stakePort int) *NodeAPIService {
```

Comment:
Can this `services.ServiceContext` be an interface in the future ?

Reply:
It's a Kurtosis-internal class so I'd like to leave it as-is, but the supersimple fix to avoid the Kurtosis coupling would be to go back to storing the ServiceID and IPAddr directly in the NodeApiService, and simply call `ServiceContext.GetServiceID()` and `ServiceContext.GetIPAddr()` at creation time (rather than inside this class)